components/list_modal.jsx

```diff
@@ -0,0 +1,234 @@
+         *
+         * Example:
+         *     const loadItems = async (pageNumber, searchTerm) => {
+         *         const {data} = await loadFromServer(searchTerm, pageNumber, PER_PAGE);
+         *         return {
+         *             items: data.users,
+         *             totalCount: data.total,
+         *         };
+         *     };
+         */
+        loadItems: PropTypes.func.isRequired,
+
+        /**
+         * renderRow is a function that receives the params (item, listModal) and should return JSX.
+         * item: an object as returned by each entry in the loadItems function's 'items' array.
+         * listModal: the instance of the ListModal component class.
+         *
+         * Example:
+         *     const renderRow = (item, listModal) => <div>{item.id}</div>;
+         */
+        renderRow: PropTypes.func.isRequired,
+
+        /**
+         * onHide (optional) a function to be invoked when the modal is closed.
+         */
+        onHide: PropTypes.func,
+
+        /**
+         * titleText (optional) a string to show at the top bar of the modal.
+         */
+        titleText: PropTypes.string,
+
+        /**
+         * searchPlaceholderText (optional) a string to show as a placeholder in the search input.
+         */
+        searchPlaceholderText: PropTypes.string,
+
+        /**
+         * titleBarButton (optional) an object representing a title button.
+         */
+        titleBarButton: PropTypes.shape({
+            text: PropTypes.string.isRequired,
+            onClick: PropTypes.func.isRequired,
+        }),
+
+        /**
+         * numPerPage (optional) a number setting how many items per page should be displayed. Defaults to
+         * DEFAULT_NUM_PER_PAGE.
+         */
+        numPerPage: PropTypes.number,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.numPerPage = props.numPerPage || DEFAULT_NUM_PER_PAGE;
+
+        this.state = {
+            show: true,
+            page: 0,
+            items: [],
+            totalCount: 0,
+            loading: true,
+            searchTerm: '',
+        };
+    }
+
+    async componentWillMount() {
+        const {totalCount, items} = await this.props.loadItems(0, '');
+        this.setState({totalCount, items, loading: false});
+    }
+
+    handleHide = () => {
+        this.setState({show: false});
+    }
+
+    handleExit = () => {
+        if (this.props.onHide) {
+            this.props.onHide();
+        }
+    }
+
+    renderRows() {
+        if (this.state.loading) {
+            return (
+                <div>
+                    <LoadingScreen
+                        position='absolute'
+                        key='loading'
+                    />
+                </div>
+            );
+        }
+        return this.state.items.map((item) => (
+            this.props.renderRow(item, this)
+        ));
+    }
+
+    onNext = async () => {
```

Comment:
I don't see the point of the `async` here

Reply:
It's not needed after I refactored this method, you're right.